docs/design/AnoncredsW3CCompatibility.md

```diff
@@ -0,0 +1,286 @@
+        rev_states: dict = None,
+    ) -> str:
+...
+...
+  try:
+    secret = await self.get_master_secret()
+    presentation = await asyncio.get_event_loop().run_in_executor(
+        None,
+        Presentation.create,
+        presentation_request,
+        present_creds,
+        self_attest,
+        secret,
+        {
+            schema_id: schema.to_native()
+            for schema_id, schema in schemas.items()
+        },
+        {
+            cred_def_id: cred_def.to_native()
+            for cred_def_id, cred_def in credential_definitions.items()
+        },
+    )
+...
+```
+
+#### Converting an already issued legacy anoncreds to W3C format(vice versa)
+
+In this case, we can use `to_w3c` method of `Credential` class to convert from legacy to w3c and `to_legacy` method of `W3CCredential` class to convert from w3c to legacy.<br>
+
+We could call `to_w3c` method like this:
+
+```
+w3c_cred = Credential.to_w3c(cred_def)
+```
+
+and for `to_legacy`:
+
+```
+legacy_cred = W3CCredential.to_legacy()
+```
+
+We don't need to input any parameters to it as it in turn calls `Credential.from_w3c()` method under the hood
+
+### How a W3C credential is stored in the wallet.
+
+Storing a credential in the wallet is somewhat dependent on the kinds of metadata that are relevent. The metadata mapping between the W3C credential and an anoncreds credential is not fully clear yet.
+
+One of the questions we need to answer is whether the preferred approach is to modify the existing store credential function so that any credential type is a valid input, or whether there should be a special function just for storing W3C credentials.
+
+We will duplicate this [store_credential](https://github.com/hyperledger/aries-cloudagent-python/blob/8cfe8283ddb2a85e090ea1b8a916df2d78298ec0/aries_cloudagent/anoncreds/holder.py#L167) function and modify it:
+
+```
+async def store_w3c_credential(...) {
+    ...
+    ...
+    try:
+        cred = W3CCredential.load(credential_data)
+    ...
+    ...
+}
+```
+
+**Question: Would it also be possible to generate the credentials on the fly to eliminate the need for storage?**
+
+### How can we handle multiple signatures on a W3C VC Format credential?
+
+Only one of the signature types (CL) is allowed in the AnonCreds format, so if a W3C VC is created by `to_legacy()`, all signature types that can't be turned into a CL signature will be dropped. This would make the conversion lossy. Similarly, an AnonCreds credential carries only the CL signature, limiting output from `to_w3c()` signature types that can be derived from the source CL signature. A possible future enhancement would be to add an extra field to the AnonCreds data structure, in which additional signatures could be stored, even if they are not used. This could eliminate the lossiness, but it adds extra complexity and may not be worth doing.
+
+### Compatibility with AFJ: how can we make sure that we are compatible?
```

Comment:
This is crucial -- especially in the design phase, so both groups are in sync, and with tests.